integreat_cms/api/v3/social_media_headers.py

```diff
@@ -0,0 +1,346 @@
+
+        :param request: Django request
+        :param \*args: The supplied arguments
+        :param \**kwargs: The supplied kwargs
+
+        :return: The response of the given function or a partial :class:`~django.http.HttpResponse`
+        """
+        try:
+            return function(request, *args, **kwargs)
+        except Http404 as e:
+            return render_error_headers(
+                request=request,
+                error=str(e),
+            )
+
+    return wrap
+
+
+def render_social_media_headers(
+    request: HttpRequest, title: str, language_code: str, excerpt: str | None, url: str
+) -> HttpResponse:
+    """
+    Renders the social media headers with the specified arguments
+
+    :param request: The current request
+    :param title: The title of the page in the social media headers
+    :param language_code: The language of the requested resource
+    :param excerpt: An optional excerpt describing the content of the page. If omitted google, will automatically crawl an excerpt
+    :param url: The url the headers belong to
+
+    :return: HTML social meta headers required by social media platforms
+    """
+    return render(
+        request,
+        "social_media_headers.html",
+        {
+            "title": title,
+            "excerpt": excerpt,
+            "url": url,
+            "image": f"{settings.BASE_URL}/{settings.SOCIAL_PREVIEW_IMAGE}",
+            "language_code": language_code,
+        },
+    )
+
+
+def render_error_headers(request: HttpRequest, error: str) -> HttpResponse:
+    """
+    Renders the partial HTML response for the webapp's server side include
+
+    :param request: The current request
+    :param error: The error message
+
+    :return: Partial HTML response for the webapp's server side include
+    """
+    return render(
+        request,
+        "error_headers.html",
+        {
+            "title": f"Error 404 | {settings.BRANDING_TITLE}",
+            "error": error,
+        },
+        status=404,
+    )
+
+
+@partial_html_response
+def root_social_media_headers(
+    request: HttpRequest, language_slug: str = settings.LANGUAGE_CODE
+) -> HttpResponse:
+    """
+    Renders the social media headers for a root page
+
+    :param request: The current request
+    :param language_slug: The language slug of the page or the default language
+
+    :return: HTML social meta headers required by social media platforms
+    """
+    language = get_object_or_404(Language, slug=language_slug)
+    title = language.social_media_webapp_title or settings.BRANDING_TITLE
+    url = site_url(request)
+
+    return render_social_media_headers(
+        request,
+        title,
+        language.bcp47_tag,
+        language.social_media_webapp_description,
+        url,
+    )
+
+
+@partial_html_response
+# pylint: disable=unused-argument
+def region_social_media_headers(
+    request: HttpRequest, region_slug: str, language_slug: str = settings.LANGUAGE_CODE
```

Comment:
Is there a reason we use the global default language here? It might make more sense to use the root language of the region

Reply:
Good point! I changed it accordingly:
https://github.com/digitalfabrik/integreat-cms/compare/b38c6e7a6adea1fb440224e5ad865b20da2192ab..5158344574eb3b9ab0f6b60b6383d3802faa6d2b